src/libraries/System.Security.Cryptography.Cng/src/System/Security/Cryptography/AesCng.cs

```diff
@@ -158,14 +161,81 @@ public override void GenerateIV()
                 iv: iv.ToArray(),
                 encrypting: false,
                 paddingMode,
-                CipherMode.CBC);
+                CipherMode.CBC,
+                feedbackSizeInBits: 0);
 
             using (transform)
             {
                 return transform.TransformOneShot(ciphertext, destination, out bytesWritten);
             }
         }
 
+        protected override bool TryDecryptCfbCore(
+            ReadOnlySpan<byte> ciphertext,
+            ReadOnlySpan<byte> iv,
+            Span<byte> destination,
+            PaddingMode paddingMode,
+            int feedbackSizeInBits,
+            out int bytesWritten)
+        {
+            ValidateCFBFeedbackSize(feedbackSizeInBits);
+
+            UniversalCryptoTransform transform = _core.CreateCryptoTransform(
+                iv: iv.ToArray(),
+                encrypting: false,
+                paddingMode,
+                CipherMode.CFB,
+                feedbackSizeInBits);
+
+            using (transform)
+            {
+                return transform.TransformOneShot(ciphertext, destination, out bytesWritten);
+            }
+        }
+
+        protected override bool TryEncryptCfbCore(
+            ReadOnlySpan<byte> plaintext,
+            ReadOnlySpan<byte> iv,
+            Span<byte> destination,
+            PaddingMode paddingMode,
+            int feedbackSizeInBits,
+            out int bytesWritten)
+        {
+            ValidateCFBFeedbackSize(feedbackSizeInBits);
+
+            UniversalCryptoTransform transform = _core.CreateCryptoTransform(
+                iv: iv.ToArray(),
+                encrypting: true,
+                paddingMode,
+                CipherMode.CFB,
+                feedbackSizeInBits);
+
+            using (transform)
+            {
+                return transform.TransformOneShot(plaintext, destination, out bytesWritten);
+            }
+        }
+
+        private void ValidateCFBFeedbackSize(int feedback)
+        {
+            if (_core.KeyInPlainText)
+            {
+                // CFB8 and CFB128 are valid for bcrypt keys.
+                if (feedback != 8 && feedback != 128)
+                {
+                    throw new CryptographicException(string.Format(SR.Cryptography_CipherModeFeedbackNotSupported, feedback, CipherMode.CFB));
```

Comment:
In the final analysis I think this is OK, but here are thoughts I had in the meantime.

* The caller specifying the feedback size seems like they probably know that the only two valid options are 8 and alg.BlockSize, so an ArgumentException seems more appropriate.
* The case of persisted keys only doing CFB8 is different, they wouldn't really know they hit a provider limitation, so that's not an ArgumentException.
* Yeah, CreateEncryptor/CreateDecryptor throws CryptographicException for this, but that's because they didn't take arguments, they took property state.
* Wait, they don't throw there, because the property validates.  So back to ArgumentException.
* Double wait, those validating properties throw CryptographicException.
* I think I might still prefer ArgumentException.  Except we'd have to figure out the model for CFB{block}+persisted CNG.
* Matching is good, though.
* Eh, it's fine how it is.

Reply:
Yeah I used the property setter as inspiration.